src/main/java/org/broadinstitute/hellbender/tools/walkers/bqsr/RecalibrationEngine.java

```diff
@@ -66,8 +67,17 @@ public void updateDataForRead( final ReadRecalibrationInfo recalInfo ) {
                     final byte qual = recalInfo.getQual(eventType, offset);
                     final double isError = recalInfo.getErrorFraction(eventType, offset);
 
+                    for (int j=0; j<2; j++) {
+                        if (keys[j]<0) {
+                            throw new GATKException.ShouldNeverReachHereException("keys["+j+"]="+keys[j]+", should not have been negative.");
+                        }
```

Comment:
yup. I can remove it from master if you want, that one isn't hard to recreate :)